### DISCUSSION TITLE
it's just too unreliable for practical use

### DISCUSSION THREAD

**Discussion by ochompsky** [tesla-android/issue-tracker#50] [General]
so i've been using this day to day for a while now and compared to the single device solution i saw on twitter, this is not the better implementation.  

this is a great proof of concept and has better performance but the double-pi solution is just too unreliable.  part of it is tesla's limitations im sure but i find having to constantly reset the device mid drive is too dangerous and just not worth the effort.

**Comment by mikegapinski** [comment]
Get back when the single board version is out, I agree with you about the inconvenience of using two computers for it. That's why it's an alpha!

**Reply by mbhforum** [reply]
One thing I've realized is Tesla's infotainment system has really improved and in a lot of ways is better than CarPlay.  Gestures require less clicks/touches and of course are more responsive. I like being able to read message on the screen. Spotify and Sirius are easier to use natively.  

The only things I really like CarPlay for is Waze, group iMessage support and Siri works better for texting rather than Tesla's voice response system.  CarPlay's apps are more limited such as searching for songs.

**Reply by ochompsky** [reply]
> One thing I've realized is Tesla's infotainment system has really improved and in a lot of ways is better than CarPlay. Gestures require less clicks/touches and of course are more responsive. I like being able to read message on the screen. Spotify and Sirius are easier to use natively.
> 
> The only things I really like CarPlay for is Waze, group iMessage support and Siri works better for texting rather than Tesla's voice response system. CarPlay's apps are more limited such as searching for songs.

not really.  spotify on tesla is buggy and missing many features from native.  bigger screen is nice but that is literally the only advantage.  

tesla's infotainment system is pretty bad for a tech car.  buggy, poor UI design, lots of wasted space, missing features everywhere.
